packages/relay-compiler/bin/RelayCompilerBin.js

```diff
@@ -339,9 +366,15 @@ const argv = yargs
       default: false,
     },
     persist: {
-      describe: 'Persist queries to srcdir/queryMap.graphql.json',
+      describe: 'Use an md5 hash as query id to replace operation text',
       type: 'boolean',
     },
+    'persist-output': {
+      describe:
+        'The json filepath where the complete query map file will be written to',
+      type: 'string',
+      default: null,
```

Comment:
Is it not possible with yargs to use a single option (`—persist`) both with and without value?

Reply:
>Is it not possible with yargs to use a single option (—persist) both with and without value?

@alloy I spent some time looking into this. If we use just a single `--persist` option then the param type will have to change to a `string` with a default of `do-not-persist`. Like so:
```js
    persist: {
      describe: 'Use an md5 hash as query id to replace operation text',
      type: 'string',
      default: 'do-not-persist',
    },
```

Then the logic in the `run` method to detect persistence will change to this:
```js
  const persist = options.persist;
  if (persist === 'do-not-persist') {
    console.log('Not persisting');
  } else if (!persist.trim()) {
    console.log(`Persisting to default output location: ${srcDir}/queryMap.graphql.json`);
  } else {
    console.log(`Persisting to custom output location ${persist}`);
  }
```

The `--help` documentation on the console will display:
```
Options:
  --persist     Use an md5 hash as query id to replace operation text
                                            [string] [default: "do-not-persist"]
// other params...
```
So it is possible to just use a single `--persist` flag, however I feel that it's doing a bit too much and is not as clean as separating the two logical actions as separate flags. The `--help` documentation in particular is confusing, because a default string of `do-not-persist` doesn't really make sense to users unless they analyse the relay-compiler code. We can change the default string to something else, but it's not really going to help because this default string is only used in the code to detect if the `--persist` flag is completely absent from the command line. Hopefully that makes sense?

Reply:
I find it unfortunate that the tooling doesn’t nicely allow for the best UI, but your explanation makes sense and I agree with your conclusion 👍

Reply:
You can remove the `type` option and then `yargs` will treat it as either a boolean if used alone or a string if passed with a value. However I agree that they should still be separate options to not overload the meaning of `--persist`.